src/styles/main.scss

```diff
@@ -1,3 +1,54 @@
+$day-size: 100px;
+
 body {
   margin: 0;
 }
+
+.calendar {
+  width: $day-size * 7 + 20px;
+  padding: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  &__day {
+    box-sizing: border-box;
+    height: $day-size;
+    width: $day-size;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    background-color: #eee;
+    border: 1px solid black;
+    transition-duration: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background: pink;
+      transform: translate(0, -20px);
+    }
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar__day:nth-child(#{$i})::before {
+    content: "#{$i}";
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+  }
+}
+
+$week-days: mon 1, tue 2, wed 3, thu 4, fri 5, sat 6, sun 7;
```

Comment:
It's difficult to find variables in your code. Keep them in one place, let's create a separate file for variables.